policy/resolver.go

```diff
@@ -1163,31 +1178,44 @@ func (cache *policyResolverCache) addCheckJobVariants(ctx context.Context, query
 
 func (cache *policyResolverCache) addDataQueryJob(ctx context.Context, query *explorer.Mquery, ownerJob *ReportingJob) {
 	uuid := cache.global.relativeChecksum(query.Mrn)
-	queryJob := cache.global.reportingJobsByUUID[uuid]
+	rj := cache.global.reportingJobsByUUID[uuid]
 
-	if queryJob == nil {
-		queryJob = &ReportingJob{
+	if rj == nil {
+		rj = &ReportingJob{
 			Uuid:       uuid,
 			QrId:       query.Mrn,
 			ChildJobs:  map[string]*explorer.Impact{},
 			Datapoints: map[string]bool{},
 			Type:       ReportingJob_DATA_QUERY,
+			Mrns:       []string{},
 			// FIXME: DEPRECATED, remove in v10.0 vv
 			DeprecatedV8IsData: true,
 		}
 		cache.global.codeIdToMrn[query.CodeId] = append(cache.global.codeIdToMrn[query.CodeId], query.Mrn)
-		cache.global.reportingJobsByUUID[uuid] = queryJob
-		cache.global.reportingJobsByMsum[query.Checksum] = append(cache.global.reportingJobsByMsum[query.Checksum], queryJob)
-		cache.childJobsByMrn[query.Mrn] = append(cache.childJobsByMrn[query.Mrn], queryJob)
+		cache.global.reportingJobsByUUID[uuid] = rj
+		cache.global.reportingJobsByMsum[query.Checksum] = append(cache.global.reportingJobsByMsum[query.Checksum], rj)
+		cache.childJobsByMrn[query.Mrn] = append(cache.childJobsByMrn[query.Mrn], rj)
+	}
+
+	if _, ok := cache.global.reportingJobsByCodeId[query.CodeId]; !ok {
```

Comment:
I'm not sure we also want to add it here.
Current focus might be only on checks.

Reply:
I think it should be fine, but please keep an eye open for side-effects (and if necessary fix this)